pkg/reconcilermanager/controllers/otel_controller_test.go

```diff
@@ -167,7 +509,19 @@ func TestOtelReconcilerGooglecloud(t *testing.T) {
 	gotConfigMap := &corev1.ConfigMap{}
 	err := fakeClient.Get(ctx, cmKey, gotConfigMap)
 	require.NoError(t, err, "ConfigMap[%s] not found", cmKey)
-	asserter.Equal(t, wantConfigMap, gotConfigMap, "ConfigMap")
+
+	// Compare metadata and labels as before
+	require.Equal(t, wantConfigMap.ObjectMeta.Labels, gotConfigMap.ObjectMeta.Labels, "ConfigMap labels")
+	require.Equal(t, wantConfigMap.ObjectMeta.Name, gotConfigMap.ObjectMeta.Name, "ConfigMap name")
+	require.Equal(t, wantConfigMap.ObjectMeta.Namespace, gotConfigMap.ObjectMeta.Namespace, "ConfigMap namespace")
+
+	// Compare the YAML content as objects, not strings
+	var wantObj, gotObj map[string]interface{}
+	require.NoError(t, yaml.Unmarshal([]byte(wantConfigMap.Data["otel-collector-config.yaml"]), &wantObj))
+	require.NoError(t, yaml.Unmarshal([]byte(gotConfigMap.Data["otel-collector-config.yaml"]), &gotObj))
+	if !reflect.DeepEqual(wantObj, gotObj) {
+		t.Errorf("ConfigMap otel-collector-config.yaml does not match:\n%s", cmp.Diff(wantObj, gotObj))
```

Comment:
What's the reason for comparing them as objects as opposed to strings? I expect the output is not as easy to read.

It would be nice if this output the actual yaml on failure to make it easier to update (copy/paste) the expected yaml string above.